src/transform/ancestor.js

```diff
@@ -118,7 +118,7 @@ function canBeLifted(doc, range) {
   }
 }
 
-// :: (Node, Pos, ?Pos) → bool
+// :: (Node, Pos, ?Pos) → ?{found: any, range: any}
```

Comment:
I'm curious about how many hacks would be needed -- I would definitely be willing to sacrifice some simplicity to get static checking. (I would not be willing to sacrifice a _lot_ of simplicity to get it, though.)